contracts/lib.rs

```diff
@@ -429,11 +462,11 @@ pub mod prosopo {
             let provider_account = self.env().caller();
             // this function is for registration only
             if self.providers.get(&provider_account).is_some() {
-                return Ok(());
+                return err!(Error::ProviderDoesNotExist);
```

Comment:
this should be `ProviderExists` or something like that

Reply:
oops, my bad! will fix :)

Reply:
fixed